src/key-container/local-storage-container.js

```diff
@@ -44,55 +46,235 @@ class LocalStorageContainer {
+  increaseKeyPath() {
+    if (this.isUnlock()) {
+      const { keySeed, pathKey } = this.getKeySeed();
+      const increasePathKey = pathKey + 1;
+      const pathKeyDb = Buffer.alloc(4);
+      pathKeyDb.writeUInt32BE(increasePathKey);
+      const keySeedEncrypted = kcUtils.encrypt(this.encryptionKey, keySeed);
+      const pathKeySeedEncrypted = kcUtils.encrypt(this.encryptionKey, utils.bytesToHex(pathKeyDb));
+      this.db.insert(`${this.prefix}keySeed`, JSON.stringify({ keySeedEncrypted, pathKeySeedEncrypted }));
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Generates recovery seed and store it into the database
+   * @param {String} masterSeed - Master seed representes as a memonic
+   * @return {String} - Public recovery address
+   */
+  generateRecoveryAddr(masterSeed) {
+    if (this.isUnlock()) {
+      const root = hdkey.fromMasterSeed(masterSeed);
+      const pathRecovery = "m/44'/60'/1'";
+      const addrNodeRecovery = root.derive(pathRecovery);
+      const privRecovery = addrNodeRecovery._privateKey;
+      const addressRecovery = ethUtil.privateToAddress(privRecovery);
+      const addressRecoveryHex = utils.bytesToHex(addressRecovery);
+      const privRecoveryHex = utils.bytesToHex(privRecovery);
+      const privRecoveryHexEncrypted = kcUtils.encrypt(this.encryptionKey, privRecoveryHex);
+      this.db.insert(this.prefix + CONSTANTS.IDRECOVERYPREFIX + addressRecoveryHex, privRecoveryHexEncrypted);
+      return addressRecoveryHex;
+    }
+    return undefined;
+  }
+
+  /**
+   * Generates a key from a given key chain and its path
+   * Key is stored as: { 'public address': 'encrypted(private key)' }
+   * @param {Number} keyChainPath - First path to derive the key
+   * @param {Number} keyPath - Second path to derive the key
+   * @returns {String} Public address key generated
+   */
+  generateSingleKey(keyProfilePath, keyPath) {
+    if (this.isUnlock()) {
+      let path = "m/44'/60'/0'/";
+      path = `${path + keyProfilePath}/${keyPath}`;
+      const { keySeed } = this.getKeySeed();
+      const root = hdkey.fromMasterSeed(keySeed);
+      const addrNode = root.derive(path);
+      const privK = addrNode._privateKey;
+      const address = ethUtil.privateToAddress(addrNode._privateKey);
+      const addressHex = utils.bytesToHex(address);
+      const privKHex = utils.bytesToHex(privK);
+      const privKHexEncrypted = kcUtils.encrypt(this.encryptionKey, privKHex);
+      this.db.insert(this.prefix + addressHex, privKHexEncrypted);
+      return addressHex;
+    }
+    return undefined;
+  }
+
+  /**
+   * Gets recovery address from the data base
+   * @return {String} - Recovery public address
+   */
+  getRecoveryAddr() {
+    if (this.isUnlock()) {
+      const idSeedKey = this.db.listKeys(this.prefix + CONSTANTS.IDRECOVERYPREFIX);
+      if (idSeedKey.length === 0) {
+        return undefined;
+      }
+      return idSeedKey[0].replace(this.prefix + CONSTANTS.IDRECOVERYPREFIX, '');
```

Comment:
This could be simplified and I'm pretty sure is faster to:

return  this.db.listKeys(this.prefix + CONSTANTS.IDRECOVERYPREFIX).length !== 0
           ? idSeedKey[0].replace(this.prefix + CONSTANTS.IDRECOVERYPREFIX, '') 
           : null

A short-hand of this could be as well:
return this.db.listKeys(this.prefix + CONSTANTS.IDRECOVERYPREFIX).length !== 0 
 && idSeedKey[0].replace(this.prefix + CONSTANTS.IDRECOVERYPREFIX, '');

If first part of the operand evaluates to true, the right part is defined, if not, a FALSE is returned.
This is not a problem... Since I guess that in the method that is calling this function will evaluate if (getRecoveryAddr()) and will be FALSE either is undefined, null or false.

When && is used, from left to right is done in the operator precedence. So if first evaluates to FALSE, is clear that won't continue evaluating nothing and will return false.

Please, for more information in JS about operands precedence check this resource:
http://www.javascriptkit.com/jsref/precedence_operators.shtml